src/Products/PythonScripts/PythonScript.py

```diff
@@ -123,6 +129,11 @@ class PythonScript(Script, Cacheable):
         Cacheable.manage_options
 
     def __init__(self, id):
+        if id in BAD_IDENTIFIERS:
+            raise ValueError(
+                '%s is not allowed as an identifier. '
+                'Please choose another name.' % id,
+                )
```

Comment:
I would indicate here that the name has special meaning for a PythonScript and can therefore not be used as identifier. I would also indicate the other simiilar names.

Reply:
Thank you - I updated the error message.